tests/integration/targets/dhcp_option_code/tasks/main.yml

```
@@ -0,0 +1,253 @@
+---
+- module_defaults:
+    group/infoblox.universal_ddi.all:
+      portal_url: "{{ portal_url }}"
+      portal_key: "{{ portal_key }}"
+  block:
+    - ansible.builtin.set_fact:
+        option_code_name: "test-basic-opt-{{ 999999 | random | string }}"
+        option_code_name2: "test-basic-opt-{{ 999999 | random | string }}"
+        option_code_name3: "test-basic-opt-{{ 999999 | random | string }}"
+        option_item_type: "boolean"
+
+    - name: Create a DHCP Option Code (check mode)
+      infoblox.universal_ddi.dhcp_option_code:
+        code: 234
+        name: "{{ option_code_name }}"
+        option_space: "{{_option_space.id}}"
+        type: "{{ option_item_type }}"
+        state: present
+      check_mode: true
+      register: option_code
+    - name: Get Information about the DHCP Option Code
+      infoblox.universal_ddi.dhcp_option_code_info:
+        filters:
+          name: "{{ option_code_name }}"
+          code: 234
+          option_space: "{{_option_space.id}}"
+      register: option_code_info
+    - assert:
+        that:
+          - option_code is changed
+          - option_code_info is not failed
+          - option_code_info.objects | length == 0
+
+    - name: Create a DHCP Option Code
+      infoblox.universal_ddi.dhcp_option_code:
+        code: 234
+        name: "{{ option_code_name }}"
+        option_space: "{{_option_space.id}}"
+        type: "{{ option_item_type }}"
+        state: present
+      register: option_code
+    - name: Get Information about the DHCP Option Code
+      infoblox.universal_ddi.dhcp_option_code_info:
+        filters:
+          name: "{{ option_code_name }}"
+          code: 234
+          option_space: "{{_option_space.id}}"
+      register: option_code_info
+    - assert:
+        that:
+          - option_code is changed
+          - option_code_info is not failed
+          - option_code_info.objects | length == 1
+          - option_code_info.objects[0].id == option_code.id
+          - option_code_info.objects[0].name == option_code.object.name
```

```suggestion
          - option_code_info.objects[0].name == option_code_name
```
Please change this for all occurences.

---

done